### PR TITLE
implement newmat/11.0.1

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **newmat/11.0.1**

implements https://github.com/conan-io/conan-center-index/issues/18989

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by AndreyMlashkin** [comment]
@ericLemanissier @uilianries 
is there some infrastructural problem right now or it's about my PR?
If I click details, all I see is "403 Forbidden"

**Comment by uilianries** [comment]
@AndreyMlashkin Please, do not close/open your PRs. The CI bot and Conan team will re-run all failed jobs affected by the current incident. As we have many PRs, enforcing by closing/opening will overload Jenkins.

**Comment by AndreyMlashkin** [comment]
What if there is no license in the repository at all?
@uilianries @SSE4

**Comment by AndreyMlashkin** [comment]
@jcar87 @RubenRBS 
What if there is no license in the repository at all?
Is there a way to suppress the error from the CI or could you point me to another recipe, having the same problem - how it is solved there?

**Comment by uilianries** [comment]
> @jcar87 @RubenRBS What if there is no license in the repository at all? Is there a way to suppress the error from the CI or could you point me to another recipe, having the same problem - how it is solved there?

@AndreyMlashkin Please read https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/conanfile_attributes.md#license-attribute

In summarize, you need to find a header or file telling about the license. If there is no hint about the recipe, then the PR can not be accepted. In case the license is really missing, please, check with the author to provide one.

**Comment by AndreyMlashkin** [comment]
> > @jcar87 @RubenRBS What if there is no license in the repository at all? Is there a way to suppress the error from the CI or could you point me to another recipe, having the same problem - how it is solved there?
> 
> @AndreyMlashkin Please read https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/conanfile_attributes.md#license-attribute
> 
> In summarize, you need to find a header or file telling about the license. If there is no hint about the recipe, then the PR can not be accepted. In case the license is really missing, please, check with the author to provide one.

okey, got it, I have written an email to the author of the code, waiting for the response

**Comment by AndreyMlashkin** [comment]
> > > @jcar87 @RubenRBS What if there is no license in the repository at all? Is there a way to suppress the error from the CI or could you point me to another recipe, having the same problem - how it is solved there?
> > 
> > 
> > @AndreyMlashkin Please read https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/conanfile_attributes.md#license-attribute
> > In summarize, you need to find a header or file telling about the license. If there is no hint about the recipe, then the PR can not be accepted. In case the license is really missing, please, check with the author to provide one.
> 
> okey, got it, I have written an email to the author of the code, waiting for the response

in the end, author pointed me to the license text

**Comment by RubenRBS** [comment]
The authentication issue has been resolved, re-triggering build :)

**Comment by AndreyMlashkin** [comment]
@RubenRBS is it now an actual error or CI/CD error?

**Comment by RubenRBS** [comment]
Seems to be a proper error now https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/18990/11-linux-gcc/newmat/11.0.1//summary.json

**Comment by AndreyMlashkin** [comment]
@RubenRBS

**Comment by AndreyMlashkin** [comment]
Is it possible, that conanio/gcc5 is a different image rather then used in CI? Is the one used in CI public?

**Comment by uilianries** [comment]
@AndreyMlashkin The image used in CCI to build any package for GCC5 is https://hub.docker.com/r/conanio/gcc5-jnlp-slave. It's the same as https://hub.docker.com/r/conanio/gcc5, but with a Jenkins agent installed. It does not change anything, so the image `conanio/gcc5` should work.

File: recipes/newmat/all/CMakeLists.txt
Comment (uilianries) [inline recipes/newmat/all/CMakeLists.txt:1]: Could you please send this CMakeLists.txt to the author? At least in case is accepted, we no longer need to maintain it.

File: recipes/newmat/all/conandata.yml
Comment (uilianries) [inline recipes/newmat/all/conandata.yml:7]: There is no reference about these patches. Please, follow  https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/conandata_yml_format.md#patches-fields and in case they really required, please, send them to the author too, so we don't need to be the maintainer.

File: recipes/newmat/all/conanfile.py
Comment (uilianries) [inline recipes/newmat/all/conanfile.py:48]: It's not supported by the project or just a recipe improvement missing? In case the project does not support it, please, update the message according, otherwise, people will provide patches for something not tested by the upstream.

File: recipes/newmat/all/conanfile.py
Comment (uilianries) [inline recipes/newmat/all/conanfile.py:56]:
```suggestion
```
This variable will not make effect, as windows-shared is not supported at all.

File: recipes/newmat/all/conanfile.py
Comment (uilianries) [inline recipes/newmat/all/conanfile.py:78]:
```suggestion
        self.cpp_info.libs = collect_libs(self)
```
Please, try to use the library name instead of `collect_libs` so people can guess the library name always.

File: recipes/newmat/all/conanfile.py
Comment (uilianries) [inline recipes/newmat/all/conanfile.py:18]:
```suggestion
    license = "LicenseRef-newmat"
```
There is no mention about a SPDX license in the `Conditions of use`:

```
I place no restrictions on the use of newmat except that I take no liability for any problems that may arise from its use, distribution or other dealings with it.

You can use it in your commercial projects (as well as your non-commercial projects).

You can make and distribute modified or merged versions. You can include parts of it in your own software.

If you distribute modified or merged versions, please make it clear which parts are mine and which parts are modified.

For a substantially modified version, simply note that it is, in part, derived from my software. A comment in the code will be sufficient.

The software is provided as is, without warranty of any kind.

Please understand that there may still be bugs and errors. Use at your own risk. I (Robert Davies) take no responsibility for any errors or omissions in this package or for any misfortune that may befall you or others as a result of your use, distribution or other dealings with it.

```

Please, always follow the license attribute according the documentation:

https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/conanfile_attributes.md#license-attribute

File: recipes/newmat/all/conanfile.py
Comment (uilianries) [inline recipes/newmat/all/conanfile.py:68]:
```suggestion
        copy(self, "nm11.htm", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
```

Please, extract the license part from that file instead, the nm11.htm is not a license file, but an entire documentation. You can use https://github.com/conan-io/conan-center-index/blob/master/recipes/angelscript/all/conanfile.py#L76 as reference.

File: recipes/newmat/all/conanfile.py
Comment (uilianries) [inline recipes/newmat/all/conanfile.py:75]:
```suggestion
            rmdir(self, os.path.join(self.package_folder, "bin"))
```
If there is no support for windows-shared, so don't need to check

File: recipes/newmat/all/test_package/test_package.cpp
Comment (uilianries) [inline recipes/newmat/all/test_package/test_package.cpp:1]: Please, simplify this file, use a single method like printing library version.

As documented (https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/test_packages.md#minimalist-source-code) the goal is testing the generated package, not running an entire example. Remind that once this code is merged, CCI will become the maintainer, so less code is always better.

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution. I have some important suggestions about this PR, after that, I could take a look in the current build error.

**Comment by uilianries** [comment]
@AndreyMlashkin Please, don't forget to sign the CLA as pointed before: https://github.com/conan-io/conan-center-index/pull/18990#issuecomment-1658582865. Otherwise, won't be possible to merge this PR.

File: recipes/newmat/all/conanfile.py
Comment (AndreyMlashkin) [reply recipes/newmat/all/conanfile.py:48]: I think, the message is quite clear, that the recipe improvement missing, otherwise, it would be "submit fix upstream".

File: recipes/newmat/all/conandata.yml
Comment (AndreyMlashkin) [reply recipes/newmat/all/conandata.yml:7]: Fixed in 6fadd505c804c514f6abf133cd5a78b847d37fef

File: recipes/newmat/all/CMakeLists.txt
Comment (AndreyMlashkin) [reply recipes/newmat/all/CMakeLists.txt:1]: Sure, I will send the final version to the author

File: recipes/newmat/all/conanfile.py
Comment (AndreyMlashkin) [reply recipes/newmat/all/conanfile.py]: fixed in e19df78a036f6a949efea8cc84ab4c08b705d929

File: recipes/newmat/all/conanfile.py
Comment (AndreyMlashkin) [reply recipes/newmat/all/conanfile.py]: fixed in e79b320667b70d1215e0816582603b8f4d5ec9cb

File: recipes/newmat/all/test_package/test_package.cpp
Comment (AndreyMlashkin) [reply recipes/newmat/all/test_package/test_package.cpp]: fixed in 2aea9744eebd36f4c03fdc26fc83efec0933766a

**Comment by AndreyMlashkin** [comment]
again, building in docker works fine for me locally...

```
docker run  -v  C:\Users\andrei\sources\conan-center-index:/sources conanio/gcc9  bash -c "conan create /sources/recipes/newmat/all newmat/11.0.1@ --build missing -s compiler.libcxx=libstdc++11 -o newmat:shared=True"
```

**Comment by AndreyMlashkin** [comment]
I have managed to reproduce the error by adding
 ```-tf .\test_v1_package\```
Could you please point me, how conan v1 test package should consume conan v2 packages?
There should be some standard way

**Comment by AndreyMlashkin** [comment]
@uilianries @SSE4 
I have added a new email to my account, I think it was the reason CI complained about me "not signing the CLA"
should I re-trigger the CI?

**Comment by AndreyMlashkin** [comment]
> @AndreyMlashkin You used a second email to push your commits, to the CLA bot, it means you need to sign the CLA with that account too, otherwise, it will not be merged as it requires CLA approved: [#18990 (comment)](https://github.com/conan-io/conan-center-index/pull/18990#issuecomment-1658582865)

I've signed it once again now

File: recipes/newmat/all/conanfile.py
Comment (uilianries) [reply recipes/newmat/all/conanfile.py:50]: the way it is written, it implies that the project supports it, but that the recipe is not prepared. Are you sure the project supports it too?

Comment (AndreyMlashkin) [reply recipes/newmat/all/conanfile.py:50]: I didn't do a research for this particular configuration (shared lib + Windows)
I just get a linker error. I am going to mark this configuration as invalid now and fix this issue later.

Comment (SSE4) [reply recipes/newmat/all/conanfile.py:50]: @uilianries it doesn't imply project supports it, how did you assume that? it just say configuration doesn't work. why exactly - requires an additional research. it might be a bug in recipe, or in conan, or in project, or in compiler, etc. this will require a significant time to figure it out on its own.

**Comment by uilianries** [comment]
@AndreyMlashkin You should sign using your another email. So far, the CLA bot didn't find all accounts signed.

**Comment by AndreyMlashkin** [comment]
> @AndreyMlashkin You should sign using your another email. So far, the CLA bot didn't find all accounts signed.

Maybe it's a bot issue?
I can only squash all commits in one using my main email, then I guess it should work

**Comment by uilianries** [comment]
@AndreyMlashkin I doubt is a bot issue, because we had similar situation in the past and we were able to solve by using both account with CLA. Squashing should work, please, let's try it then.

**Comment by AndreyMlashkin** [comment]
@SSE4 @uilianries @prince-chrismc 
please approve

**Review by SSE4 — APPROVED** [review]

File: recipes/newmat/all/conanfile.py
Comment (uilianries) [inline recipes/newmat/all/conanfile.py:50]:
```suggestion
        if is_msvc(self) and self.options.shared:
            raise ConanInvalidConfiguration(f"{self.ref} does not support dynamic library with msvc.")
```

File: recipes/newmat/all/conanfile.py
Comment (uilianries) [inline recipes/newmat/all/conanfile.py:91]:
```suggestion
        self.cpp_info.libs = ["newmat"]
```

Let's simplify it

File: recipes/newmat/all/test_package/CMakeLists.txt
Comment (uilianries) [inline recipes/newmat/all/test_package/CMakeLists.txt:4]:
```suggestion
find_package(newmat REQUIRED)
```
case sensitive

File: recipes/newmat/all/test_package/test_package.cpp
Comment (uilianries) [inline recipes/newmat/all/test_package/test_package.cpp:1]: it could be a C file instead

File: recipes/newmat/all/test_v1_package/CMakeLists.txt
Comment (uilianries) [inline recipes/newmat/all/test_v1_package/CMakeLists.txt:1]: Please, remove the test_v1_package folder. For more info, read https://github.com/conan-io/conan-center-index/discussions/17406

File: recipes/newmat/all/CMakeLists.txt
Comment (uilianries) [inline recipes/newmat/all/CMakeLists.txt:64]:
```suggestion
```
Let self.options.fPIC decides that behavior.

File: recipes/newmat/all/CMakeLists.txt
Comment (uilianries) [inline recipes/newmat/all/CMakeLists.txt:23]:
```suggestion
set(lib_name ${PROJECT_NAME})
```

The file `nm_gnu` uses `libnewmat.a` as library name.

File: recipes/newmat/all/conanfile.py
Comment (uilianries) [inline recipes/newmat/all/conanfile.py:96]:
```suggestion
        self.cpp_info.includedirs.append(os.path.join("include", "newmat"))
        if self.options.with_c_subscripts:
            self.cpp_info.defines = ["SETUP_C_SUBSCRIPTS"]
```
That definition is present in public headers

**Review by uilianries — CHANGES_REQUESTED** [review]
Thank you for your contribution, please, check my review

File: recipes/newmat/all/conandata.yml
Comment (AndreyMlashkin) [reply recipes/newmat/all/conandata.yml:7]: Would you prefer just "compilation fix for c++11 compatible compilers"

File: recipes/newmat/all/CMakeLists.txt
Comment (uilianries) [reply recipes/newmat/all/CMakeLists.txt]: @AndreyMlashkin please, revert it, the library name should be `newmat`, not `Newmat` it's important keeping the same name provided by the upstream:

nm_gnu.mak:16

```
libnewmat.a:   	$(newmat_lobj)
		$(AR) -cr $@ $(newmat_lobj)
		ranlib $@
```

File: recipes/newmat/all/conanfile.py
Comment (uilianries) [reply recipes/newmat/all/conanfile.py:90]: Once `newmat` (not `Newmat`) is used as library name, it should work.

File: recipes/newmat/all/CMakeLists.txt
Comment (uilianries) [inline recipes/newmat/all/CMakeLists.txt:4]:
```suggestion
  "newmat"
```

**Review by SSE4 — APPROVED** [review]

**Comment by AndreyMlashkin** [comment]
@uilianries please approve

File: recipes/newmat/all/conanfile.py
Comment (RubenRBS) [inline recipes/newmat/all/conanfile.py:20]: Duplicated settings, unneeded topic

```suggestion
    name = "newmat"
    description = "Manipulate a variety of types of matrices using standard matrix operations."
    url = "https://github.com/conan-io/conan-center-index"
    homepage = "https://newmat.net"
    topics = ("matrix")
    license = "LicenseRef-newmat"
    package_type = "library"
    settings = "os", "arch", "compiler", "build_type"

```

File: recipes/newmat/all/conanfile.py
Comment (RubenRBS) [inline recipes/newmat/all/conanfile.py:49]: Weird that this has a `shared` check. Does this maybe belong to `validate()` instead of `validate_build()`?

**Review by RubenRBS — COMMENTED** [review]
Thanks a lot, I have a few comments, and I'll double check tomorrow with @uilianries about the `validate_build()` comment just in case Im' wrong, which is totally possible :)

File: recipes/newmat/all/conanfile.py
Comment (AndreyMlashkin) [reply recipes/newmat/all/conanfile.py]: corrected that now

File: recipes/newmat/all/test_package/test_package.cpp
Comment (uilianries) [inline recipes/newmat/all/test_package/test_package.cpp:6]:
```suggestion
```
The source in test package should be configured or from the package or by the own test package. In this case, the definition `use_namespace` is not configurable and has no effect. Please, remove it.

**Review by uilianries — APPROVED** [review]
LGTM!

**Comment by uilianries** [comment]
@AndreyMlashkin again the CLA bot is complaining that you are mixing GH accounts. It seems like you are using 2 names: `Andrei` and `Andrey`. In case you have a second account, please, assign using both accounts, otherwise, you will need to git squash again.

**Pull request merged**
